sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/PruneFileSourcePartitionsSuite.scala

```diff
@@ -42,35 +43,27 @@ class PruneFileSourcePartitionsSuite extends PrunePartitionSuiteBase {
 
   test("PruneFileSourcePartitions should not change the output of LogicalRelation") {
     withTable("test") {
-      withTempDir { dir =>
-        sql(
-          s"""
-            |CREATE EXTERNAL TABLE test(i int)
-            |PARTITIONED BY (p int)
-            |STORED AS parquet
-            |LOCATION '${dir.toURI}'""".stripMargin)
-
-        val tableMeta = spark.sharedState.externalCatalog.getTable("default", "test")
-        val catalogFileIndex = new CatalogFileIndex(spark, tableMeta, 0)
-
-        val dataSchema = StructType(tableMeta.schema.filterNot { f =>
-          tableMeta.partitionColumnNames.contains(f.name)
-        })
-        val relation = HadoopFsRelation(
-          location = catalogFileIndex,
-          partitionSchema = tableMeta.partitionSchema,
-          dataSchema = dataSchema,
-          bucketSpec = None,
-          fileFormat = new ParquetFileFormat(),
-          options = Map.empty)(sparkSession = spark)
-
-        val logicalRelation = LogicalRelation(relation, tableMeta)
-        val query = Project(Seq(Symbol("i"), Symbol("p")),
-          Filter(Symbol("p") === 1, logicalRelation)).analyze
-
-        val optimized = Optimize.execute(query)
-        assert(optimized.missingInput.isEmpty)
-      }
+      spark.range(10).selectExpr("id", "id % 3 as p").write.partitionBy("p").saveAsTable("test")
```

Comment:
As you know, `saveAsTable` is different from `STORED AS` parquet. The original test coverage seems to be coupled with `convertMetastoreParquet`, but this one looks different. Are we losing the existing test coverage?

Reply:
This specific test coverage should remain at `hive` module.

Reply:
Hmm what is `convertMetastoreParquet`? I couldn't find it anywhere.

Regarding the test, I think it is still covered (I've debugged the test and made sure it is still going through the related code paths in `PruneFileSourcePartitions`). Much has changed since 2016 though: the test (added in #15569) was originally designed to make sure that `LogicalRelation.expectedOutputAttributes` was correctly populated in the class. The `expectedOutputAttributes`, however, was later replaced by directly passing `output` in `LogicalRelation` (in #17552), which I think further prevented the issue from happening.

Reply:
I mean `spark.sql.hive.convertMetastoreParquet`. Here is the document.
- https://spark.apache.org/docs/latest/sql-data-sources-parquet.html#hive-metastore-parquet-table-conversion

Reply:
FYI, `CREATE TABLE ... USING PARQUET` (spark syntax) and `CREATE TABLE ... STORED AS PARQUET` (hive syntax) generates different tables in Apache Spark.

Reply:
For Hive tables generated by `STORED AS` syntax, Spark converts it to data source tables on the fly because `spark.sql.hive.convertMetastoreParquet` is `true` by default. It's the same for ORC. For ORC, we have `spark.sql.hive.convertMetastoreOrc`.

Reply:
Thanks @dongjoon-hyun . I found it now. However I'm not sure whether this matters for the test though: what it does is just 1) register table metadata in the catalog, 2) create a `LogicalRelation` wrapping a `HadoopFsRelation` which has the data and partition schema from the step 1), and 3) feed it into the rule `PruneFileSourcePartitions` and see if the `LogicalRelation`'s `expectedOutputAttributes` is properly set. Seems this is irrelevant to what SerDe it is using?

Reply:
If you are not sure, why don't we keep the AS-IS one, @sunchao ?